src/django/pfb_analysis/views.py

```diff
@@ -106,33 +114,25 @@ def create(self, request, *args, **kwargs):
 
         """
         file_obj = request.data['file']
-        tmpdir = tempfile.mkdtemp()
-        try:
-            # Awkward. create_from_shapefile opens with fiona, so we can't just directly pass
-            #   in the file-like upload object. First we have to write it to disk, then pass
-            #   the on-disk location to the create method instead.
-            # Alternatively we could manually write to disk here, unzip and pass the fiona handle
-            #   to create_from_shapefile, but that duplicates tested logic we do anyways within
-            #   that method.
-            upload_filename = os.path.join(tmpdir, 'upload.zip')
-            with open(upload_filename, 'wb') as upload_file:
-                upload_file.write(file_obj.read())
-
-            try:
-                batch = AnalysisBatch.objects.create_from_shapefile(upload_filename,
-                                                                    submit=True,
-                                                                    user=request.user)
-            except Exception as e:
-                # Errors are likely to be from bad input files, so send the exception message
-                # back in an error response.
-                raise APIException(e)
-
-            # Rather than return the AnalysisBatch object which doesn't really have any useful
-            # info of its own, serialize and return the list of newly created jobs.
-            serializer = AnalysisJobSerializer(batch.jobs, many=True)
-            return Response(serializer.data, status=status.HTTP_201_CREATED)
-        finally:
-            shutil.rmtree(tmpdir, ignore_errors=True)
+
+        client = boto3.client('s3', config=BotocoreClientConfig(signature_version='s3v4'))
+
+        organization = request.user.organization
+        file_name = '{}.zip'.format(str(uuid4()))
+        key = get_batch_shapefile_upload_path(organization.name, file_name).lstrip('/')
+
+        response = client.upload_fileobj(file_obj, settings.AWS_STORAGE_BUCKET_NAME, key)
+        print(response)
+        url = client.generate_presigned_url(
+            ClientMethod='get_object',
+            Params={'Bucket': settings.AWS_STORAGE_BUCKET_NAME, 'Key': key}
+        )
+        async('pfb_analysis.tasks.create_batch_from_remote_shapefile', url)
+
+        return Response({
+            'shapefile_url': url,
+            'status': 'STARTED'
+        }, status=status.HTTP_200_OK)
```

Comment:
Yeah, that was my initial argument for a 200 instead of 201. We don't expose the AnalysisBatch object anywhere, and it might as well not exist as far as the API/client is concerned. But we did create something. I don't feel strongly about it either way, happy to change it if desired.

Reply:
Oh hey, [RFC 2616](https://tools.ietf.org/html/rfc2616#section-10.2.2) has an opinion on the matter:
> If the action cannot be carried out immediately, the server SHOULD respond with 202 (Accepted) response instead.

But the later [RFC 7231](https://tools.ietf.org/html/rfc7231#section-6.3.2) is less insistent.  Looks like in any case it's within the range of intended uses for 200, so that works fine.  (Or we could switch to 202 just to be different.  But yeah, 200 is good.)